app/models/product_test.rb

```diff
@@ -108,10 +108,9 @@ def sample_and_duplicate_records(recs, ids, random: Random.new)
     dups = records.find(ids)
 
     recs, dups = randomize_clinical_data(recs, dups, random)
-
-    (random.rand(3) + 1).times do
+    dups.sample(random.rand(3) + 1, random: random).each do |rec|
```

Comment:
If what you want is a random number in the range [1, 4]
`random.rand(3) + 1` and `random.rand(1..4)` will both do that.

If that's not what you want, then what you have here is wrong.